ec2rlcore/s3upload.py

```diff
@@ -104,37 +86,30 @@ def get_presigned_url(uploader_url, filename, region="us-east-1"):
     if region not in s3uploader_regions:
         region = "us-east-1"
 
-    query_str = urlparse.urlparse(uploader_url).query
-    put_dict = urlparse.parse_qs(query_str)
-
-    # If uploader_url is not parsable then maybe it is a shortened URL
     try:
-        if not put_dict:
-            the_request = urllib_request.Request(uploader_url)
-            uploader_url = urllib_urlopen(the_request).geturl()
-            query_str = urlparse.urlparse(uploader_url).query
-            put_dict = urlparse.parse_qs(query_str)
-    except urllib_urlerror:
-        pass
+        # Grabs the real URL if it has been shortened
+        uploader_url = requests.get(uploader_url).url
+        query_str = requests.utils.urlparse(uploader_url).query
+        put_dict = dict(x.split('=') for x in query_str.split('&'))
+    except requests.exceptions.Timeout:
+        raise requests.exceptions.Timeout("ERROR: Connection timed out.")
```

Comment:
It is redundant to catch and re-raise the same exception without adding any additional value.